lib/deimos/utils/schema_class.rb

```diff
@@ -10,10 +10,23 @@ class << self
         # @return [Array<String>]
         def modules_for(namespace)
           modules = ['Schemas']
-          namespace_folder = namespace.split('.').last
-          if Deimos.config.schema.generate_namespace_folders && namespace_folder
-            modules.push(namespace_folder.underscore.classify)
+
+          if Deimos.config.schema.generate_namespace_folders
+            namespace_map = Deimos.config.schema.schema_namespace_map.sort_by{|k,v| -k.length}.to_h
+            namespace_override = namespace_map.keys.detect{ |k| namespace.include?(k) }
```

Comment:
Also - this indicates that the namespace map is actually a map of namespace **prefixes** rather than fully articulated namespaces. If that's what's intended, we should make sure the documentation is clear about this. Or even rename this to `namespace_prefix_map`.

Reply:
good point thanks, will update the docs